x-pack/plugins/apm/server/routes/alerts/register_transaction_duration_alert_type.ts

```diff
@@ -147,17 +148,27 @@ export function registerTransactionDurationAlertType({
             },
           },
           aggs: {
-            latency:
-              ruleParams.aggregationType === 'avg'
-                ? { avg: { field } }
-                : {
-                    percentiles: {
-                      field,
-                      percents: [
-                        ruleParams.aggregationType === '95th' ? 95 : 99,
-                      ],
-                    },
-                  },
+            environments: {
+              terms: {
+                field: SERVICE_ENVIRONMENT,
+                missing: ENVIRONMENT_NOT_DEFINED.value,
+              },
+              aggs: {
+                ...(ruleParams.aggregationType === 'avg'
```

Comment:
With the helper mentioned

```suggestion
            ...averageOrPercentileAgg({ aggregationType, transactionDurationField })
```